server/src/routes.py

```diff
@@ -18,22 +22,73 @@
 def __replace_config_data(data: bytes) -> bytes:
     config = cfg()
 
+    # SIGNUP CONFIGURATION
     if not config.general.allow_signups:
         data = data.replace(b'name="PA-signup" content="true"', b'name="PA-signup" content="false"')
-    if not config.mail or not config.mail.enabled:
+    
+    # OIDC CONFIGURATION - Check first to determine if it's available
+    oidc_fully_configured = (
+        config.oidc.enabled and 
+        config.oidc.client_id and 
+        config.oidc.client_secret and
+        bool(config.oidc.domain or config.oidc.authorize_url)  # Either domain for discovery OR direct URLs
```

Comment:
These should just become mandatory in the config. If you specify an oidc config, it should contain all the data required.

For the domain vs authorize_url we probably should ideally check that in a sort of post-validation step in the config manager itself, but I'm ok with keeping it here for now